java/src/Ice/src/main/java/com/zeroc/Ice/_ObjectPrxI.java

```diff
@@ -12,6 +12,15 @@
 
 /** Concrete proxy implementation. */
 public class _ObjectPrxI implements ObjectPrx, java.io.Serializable {
+  // TODO: delete this default constructor eventually.
+  public _ObjectPrxI() {}
+
+  public _ObjectPrxI(ObjectPrx obj) {
```

Comment:
Should this constructor be protected instead, like in C#?

Reply:
At the moment, this constructor could be `protected`, but I'd like to keep it public because `ProxyFactory` will probably need it in the future.

Right now it's calling the default constructor and patching fields manually. When I remove the default constructor, it will need to use this constructor instead.